admin_guide/upgrading_citus.rst

```diff
@@ -7,67 +7,242 @@ $$$$$$$$$$$$$$$
 
 Upgrading Citus Versions
 ########################
-Upgrading the Citus version requires first obtaining the new Citus extension and
-then installing it in each of your database instances. The first step varies by
-operating system.
 
-.. _upgrading_citus_package:
+Citus adheres to `semantic versioning <http://semver.org/>`_ with patch-, minor-, and major-versions. The upgrade process differs for each, requiring more effort for bigger version jumps.
 
-Step 1. Update Citus Package
-----------------------------
-
-**OS X**
+Upgrading the Citus version requires first obtaining the new Citus extension and then installing it in each of your database instances. Citus uses separate packages for each minor version to ensure that running a default package upgrade will provide bug fixes but never break anything. Let's start by examining patch upgrades, the easiest kind.
 
-::
+Patch Version Upgrade
+---------------------
 
-  brew update
-  brew upgrade citus
+To upgrade a Citus version to its latest patch, issue a standard upgrade command for your package manager. Assuming version 6.1 is currently installed:
 
 **Ubuntu or Debian**
 
-::
+.. code-block:: bash
 
   sudo apt-get update
-  sudo apt-get upgrade postgresql-9.5-citus
+  sudo apt-get upgrade postgresql-9.6-citus-6.1
 
 **Fedora, CentOS, or Red Hat**
 
-::
+.. code-block:: bash
 
-  sudo yum update citus_95
+  sudo yum update citus61_96
 
-.. _upgrading_citus_extension:
+Then restart PostgreSQL and the upgrade will be complete.
 
-Step 2. Apply Update in DB
---------------------------
+.. _major_minor_upgrade:
 
-Restart PostgreSQL:
+Major and Minor Version Upgrades
+--------------------------------
 
-::
+Major and minor version upgrades follow the same steps, but be careful: major upgrades can make backward-incompatible changes in the Citus API. It is best to review the Citus `changelog <https://github.com/citusdata/citus/blob/master/CHANGELOG.md>`_ before a major upgrade and look for any changes which may cause problems for your application.
 
-  pg_ctl -D /data/base -l logfile restart
+Each major and minor version of Citus is published as a package with a separate name. Installing a newer package will automatically remove the older version. Here is how to upgrade from 6.0 to 6.1 for instance:
 
-::
+Step 1. Update Citus Package
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-  # after restarting postgres
-  psql -c "ALTER EXTENSION citus UPDATE;"
+**Ubuntu or Debian**
+
+.. code-block:: bash
 
-  psql -c "\dx"
-  # you should see a newer Citus 6.0 version in the list
+  sudo apt-get update
+  sudo apt-get install postgresql-9.6-citus-6.1
 
-That's all it takes! No further steps are necessary after updating
-the extension on all database instances in your cluster.
 
+**Fedora, CentOS, or Red Hat**
+
+.. code-block:: bash
 
+  # Fedora, CentOS, or Red Hat
+  sudo yum install citus61_96
```

Comment:
When would `yum swap` be unavailable? Should we just document yum swap first, and then offer the alternate instructions if contacted?